hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/ZookeeperGroupRepository.java

```diff
@@ -108,4 +110,11 @@ private Optional<Group> getGroupDetails(String groupName, boolean quiet) {
         String path = paths.groupPath(groupName);
         return readFrom(path, Group.class, quiet);
     }
+
+    @PostConstruct
+    public void init() {
+        logger.info("Before ensuring init path exists");
+        ensurePathExists(paths.groupsPath());
```

Comment:
**issue:** `ensurePathExists` which is used here is _"different"_ from the original one.
You're using method from `ZookeeperBasedRepository`, where previous code was using `ZookeeperClient` class method with same name.

I quickly looked on both implementations and they look similar:

***ZookeeperClient (previous)***
```java
public void ensurePathExists(String path) {
    try {
        if (curatorFramework.checkExists().forPath(path) == null) {
            logger.info("Creating path: {} in Zookeeper: {}", path, datacenterName);
            curatorFramework.create().creatingParentsIfNeeded().forPath(path);
        }
    } catch (Exception e) {
        logger.error("Error when creating path: {} in Zookeeper: {}", path, datacenterName, e);
        throw new InternalProcessingException("Could not ensure existence of path: " + path);
    }
}
```

***ZookeeperBasedRepository (current)***
```java
protected void ensurePathExists(String path) {
    ensureConnected();
    if (!pathExists(path)) {
        try {
            zookeeper.create().creatingParentsIfNeeded().forPath(path);
        } catch (Exception e) {
            throw new InternalProcessingException(e);
        }
    }
}
```

**suggestion:** Ensure that changing this method won't have negative effect, and if so - remove `ZookeeperClient.ensurePathExists` method, as it seems not used nowhere else.

Reply:
Underneath they do the same thing: check if path exists using curator framework and create the path with parents if needed. 

If we were to comment out this check e2e tests would fail immediately:

```
Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /hermes/groups
```

Removed the `ZookeeperClient` version since its no longer used.